jenkins/keti/Jenkinsfile

```diff
@@ -1,106 +1,158 @@
+                        sh "cd public-kt/keti/kotlin && mvn clean package install ${mavenOpts} -P public -D maven.repo.local=${env.WORKSPACE}/m2-public-kt"
+                    }
+                }
+                stage("Build the 'public-graph' Maven profile (Java)") {
+                    steps {
+                        sh "cd public-graph/keti && mvn clean package install ${mavenOpts} -P public-graph -D maven.repo.local=${env.WORKSPACE}/m2-public-graph"
+                    }
+                }
+                stage("Build the 'public-graph' Maven profile (Kotlin)") {
+                    steps {
+                        sh "cd public-graph-kt/keti/kotlin && mvn clean package install ${mavenOpts} -P public-graph -D maven.repo.local=${env.WORKSPACE}/m2-public-graph-kt"
+                    }
+                }
+            }
+            post {
+                always {
+                    junit testResults: '**/public*/**/surefire-reports*/junitreports/TEST*.xml, **/public*/**/surefire-reports*/**/junitreports/TEST*.xml', allowEmptyResults: true
+                }
+            }
+        }
+        stage('Run integration tests') {
+            failFast true
+            parallel {
+                stage("Run integration tests using the 'public' Maven profile (Java)") {
+                    steps {
+                        sh "cd public/keti && PORT_OFFSET=10 source ./set-env-local.sh && cd acs-integration-tests && mvn clean verify ${mavenOpts} -P public -D maven.repo.local=${env.WORKSPACE}/m2-public"
+                    }
+                }
+                stage("Run integration tests using the 'public' Maven profile (Kotlin)") {
+                    steps {
+                        sh "cd public-kt/keti/kotlin && PORT_OFFSET=20 source ./set-env-local.sh && cd acs-integration-tests && mvn clean verify ${mavenOpts} -P public -D maven.repo.local=${env.WORKSPACE}/m2-public-kt"
+                    }
+                }
+                stage("Run integration tests using the 'public-graph' Maven profile (Java)") {
+                    steps {
+                        sh "cd public-graph/keti && PORT_OFFSET=30 source ./set-env-local.sh && cd acs-integration-tests && mvn clean verify ${mavenOpts} -P public-graph -D maven.repo.local=${env.WORKSPACE}/m2-public-graph"
+                    }
+                }
+                stage("Run integration tests using the 'public-graph' Maven profile (Kotlin)") {
+                    steps {
+                        sh "cd public-graph-kt/keti/kotlin && PORT_OFFSET=40 source ./set-env-local.sh && cd acs-integration-tests && mvn clean verify ${mavenOpts} -P public-graph -D maven.repo.local=${env.WORKSPACE}/m2-public-graph-kt"
+                    }
+                }
+            }
+            post {
+                always {
+                    junit testResults: '**/public*/**/failsafe-reports*/junitreports/TEST*.xml, **/public-*/**/failsafe-reports*/**/junitreports/TEST*.xml', allowEmptyResults: true
```

Comment:
@anubhavi25, I noticed that we don't separate test results.  Only the result of the latest run is recorded on Jenkins dashboard and "Latest Test Results".  Is it possible to record test results per profile?  Since tests are running in parallel, it won't be easy to tell for what profile test failed, will it?

Note, that this is not a problem introduced by this commit.  We were not separating test results in prior revisions including predix.  We could handle it as a separate enhancement story.

Reply:
I can look into this as a later enhancement but for now you can view the logs in the [staged view](https://ci.eclipse.org/keti/job/Keti-Build/job/PR-23/workflow-stage/) for each individual stage by clicking the colored rectangle under the stage's title and then clicking "Logs"